Error: The argument type 'String?' can't be assigned to the parameter type 'String'

**Describe the bug**

I have updated some packages and run `flutter clean` then when try to run the app I got this error below, the app before the update was working fine 

The pubspec

```yamel
name: masarifi_app
description: Masarifi app for reviewing your expenses on a regular basis.

# The following line prevents the package from being accidentally published to
# pub.dev using `flutter pub publish`. This is preferred for private packages.
publish_to: "none" # Remove this line if you wish to publish to pub.dev

version: 3.8.8+65

environment:
  sdk: ">=3.1.2 <4.0.0"

dependencies:
  animated_bottom_navigation_bar: ^1.0.1
  auto_route: ^7.7.1
  auto_size_text: ^3.0.0
  community_charts_flutter: ^1.0.4
  countup: ^0.1.4
  cupertino_icons: ^1.0.2
  dartz: ^0.10.1
  datetime_picker_formfield: ^2.0.0
  device_info_plus: ^9.0.2
  dio: ^5.3.0
  drift: ^2.20.0
  drift_sqflite: ^2.0.1
  dropdown_search: ^5.0.6
  equatable: ^2.0.5
  file: ^6.1.4
  firebase_analytics: ^10.4.4
  firebase_core: ^2.15.0
  firebase_crashlytics: ^3.3.4
  flutter:
    sdk: flutter
  flutter_bloc: ^8.1.1
  flutter_contacts: ^1.1.7
  flutter_easyloading: ^3.0.5
  flutter_expandable_fab: ^1.6.0
  flutter_facebook_auth: ^6.0.0
  flutter_hooks: ^0.19.0
  flutter_image_slideshow: ^0.1.6
  flutter_localizations:
    sdk: flutter
  flutter_native_splash: ^2.2.17
  flutter_rating_stars: ^1.0.3+4
  flutter_signin_button: ^2.0.0
  flutter_simple_calculator: ^2.2.1
  flutter_slidable: ^3.1.1
  flutter_speed_dial: ^7.0.0
  flutter_staggered_animations: ^1.0.0
  flutter_svg: ^2.0.7
  freezed_annotation: ^2.1.0
  geolocator: ^9.0.2
  get_it: ^7.2.0
  google_fonts: ^5.1.0
  google_mobile_ads: ^3.0.0
  google_sign_in: ^6.1.4
  grouped_list: ^5.1.2
  image_picker: ^1.0.1
  in_app_purchase: ^3.2.0
  injectable: ^2.1.2
  intl: ^0.18.0
  ionicons: ^0.2.2
  json_annotation: ^4.7.0
  local_auth: ^2.0.2
  onesignal_flutter: ^3.5.1
  open_filex: ^4.1.1
  path: ^1.8.0
  path_provider: ^2.0.10
  pdf: ^3.8.2
  permission_handler: ^10.2.0
  responsive_framework: ^0.2.0
  rxdart: ^0.27.3
  shared_preferences: ^2.0.11
  skeletons: ^0.0.3
  sqlite3_flutter_libs: ^0.5.24
  unicons: ^2.1.0
  url_launcher: ^6.0.20
  uuid: ^3.0.6

dependency_overrides:
  in_app_purchase_android: 0.3.6+5
  intl: ^0.18.0

dev_dependencies:
  auto_route_generator: ^7.2.0
  build_runner: ^2.4.11
  change_app_package_name: ^1.0.0
  drift_dev: ^2.20.1
  flutter_lints: ^2.0.1
  flutter_test:
    sdk: flutter
  freezed: ^2.4.1
  injectable_generator: ^2.1.6
  json_serializable: ^6.7.1

flutter_native_splash:
  color: "#4285F5"
  image: assets/images/global/transparent-logo.png
  # branding: assets/images/global/splash.png
  # color_dark: "#121212"
  # image_dark: assets/logo-development.png
  # branding_dark: assets/branding-development.png
  android_12:
    image: assets/images/global/transparent-logo.png
    color: "#4285F5"
    # image_dark: assets/logo-development.png
    # icon_background_color_dark: "#121212"

  # The android, ios and web parameters can be used to disable generating a splash screen on a given
  # platform.
  android: true
  ios: false
  web: false

  fullscreen: true
  android_gravity: center

flutter:
  uses-material-design: true
  assets:
    - assets/i18n/
    - assets/data/
    - assets/fonts/HacenTunisia/
    - assets/images/categories/
    - assets/images/global/
    - google_fonts/
  #   - family: Trajan Pro
  #     fonts:
  #       - asset: fonts/TrajanPro.ttf
  #       - asset: fonts/TrajanPro_Bold.ttf
  #         weight: 700
  #
  # For details regarding fonts from package dependencies,
  # see https://flutter.dev/custom-fonts/#from-packages

```

The Error: 

```dash
Launching lib/main.dart on sdk gphone64 arm64 in debug mode...
You are applying Flutter's app_plugin_loader Gradle plugin imperatively using the apply script method, which is deprecated and will be removed in a future release. Migrate to applying Gradle plugins with the declarative plugins block: https://flutter.dev/go/flutter-gradle-plugin-apply

You are applying Flutter's main Gradle plugin imperatively using the apply script method, which is deprecated and will be removed in a future release. Migrate to applying Gradle plugins with the declarative plugins block: https://flutter.dev/go/flutter-gradle-plugin-apply

lib/core/infrastructure/datasources/database/database.g.dart:2574:47: Error: The argument type 'String?' can't be assigned to the parameter type 'String' because 'String?' is nullable and 'String' isn't.
        .filter((f) => f.categoryId.id($_item.id));
                                              ^
lib/core/infrastructure/datasources/database/database.g.dart:2823:43: Error: The argument type 'int?' can't be assigned to the parameter type 'int' because 'int?' is nullable and 'int' isn't.
        .filter((f) => f.userId.id($_item.id));
                                          ^
lib/core/infrastructure/datasources/database/database.g.dart:3241:45: Error: The argument type 'String?' can't be assigned to the parameter type 'String' because 'String?' is nullable and 'String' isn't.
        .filter((f) => f.walletId.id($_item.id));
                                            ^
Target kernel_snapshot failed: Exception


FAILURE: Build failed with an exception.

* Where:
Script '/Users/heshamshawky/Developer/flutter/packages/flutter_tools/gradle/src/main/groovy/flutter.groovy' line: 1629

* What went wrong:
Execution failed for task ':app:compileFlutterBuildDebug'.
> Process 'command '/Users/heshamshawky/Developer/flutter/bin/flutter'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 4s
Error: Gradle task assembleDebug failed with exit code 1
```

--- a/drift/lib/src/runtime/manager/filter.dart
+++ b/drift/lib/src/runtime/manager/filter.dart
@@ -71,10 +71,11 @@ class ColumnFilters<T extends Object> extends _BaseColumnFilters<T> {
       ColumnFilters(column, inverted: !inverted, joinBuilders: joinBuilders);
 
   /// Create a filter that checks if the column equals a value.
-  ComposableFilter equals(T value) => $composableFilter(column.equals(value));
+  ComposableFilter equals(T? value) =>
+      $composableFilter(column.equalsNullable(value));
 
   /// Shortcut for [equals]
-  ComposableFilter call(T value) => equals(value);
+  ComposableFilter call(T? value) => equals(value);
 
   /// Create a filter that checks if the column is in a list of values.
   ComposableFilter isIn(Iterable<T> values) =>

--- a/drift/test/manager/manager_filter_test.dart
+++ b/drift/test/manager/manager_filter_test.dart
@@ -408,4 +408,13 @@ void main() {
             .count(),
         1);
   });
+
+  test('can use shorthand filter for nulls', () async {
+    final row = await db.todosTable.insertReturning(
+        TodosTableCompanion.insert(content: 'my test content'));
+
+    final query =
+        await db.managers.todosTable.filter((f) => f.targetDate(null)).get();
+    expect(query, [row]);
+  });
 }

